src/task_manager.cc

```diff
@@ -237,12 +236,12 @@ namespace detail {
 			std::lock_guard lock(task_mutex);
 			current_horizon_critical_path_length = max_pseudo_critical_path_length;
 			const auto previous_horizon = current_horizon;
-			current_horizon = reduce_execution_front(std::move(reserve), task::make_horizon_task(tid));
+			current_horizon = reduce_execution_front(std::move(reserve), task::make_horizon_task(tid)).get_id();
 			if(previous_horizon) { set_epoch_for_new_tasks(*previous_horizon); }
 		}
 
 		// it's important that we don't hold the lock while doing this
-		invoke_callbacks(tid);
+		invoke_callbacks(get_task(tid));
```

Comment:
Could get rid of this `get_task` by storing the task returned by `reduce_execution_front`.

Reply:
Done!